komga-webui/src/components/dialogs/EditSeriesDialog.vue

```diff
@@ -611,6 +656,31 @@ export default Vue.extend({
       return null
     },
     async editSeries(): Promise<boolean> {
+      if (this.single && this.uploadQueue.length > 0) {
+        const name = this.selectedThumbnail
+        const series = this.series as SeriesDto
+        for (const file of this.uploadQueue) {
+          const body = new FormData()
+          body.append('file', file)
+          body.append('selected', `${file.name === name}`)
+          await this.$komgaSeries.uploadThumbnail(series.id, body)
```

Comment:
Moving the building of the `FormData` into the service would clarify the intent of the `uploadThumbnail` function and hide underlying complexity in the service